src/main/kotlin/io/libp2p/security/noise/NoiseXXSecureChannel.kt

```diff
@@ -0,0 +1,315 @@
+package io.libp2p.security.noise
+
+import com.google.protobuf.ByteString
+import com.southernstorm.noise.protocol.CipherState
+import com.southernstorm.noise.protocol.CipherStatePair
+import com.southernstorm.noise.protocol.DHState
+import com.southernstorm.noise.protocol.HandshakeState
+import com.southernstorm.noise.protocol.Noise
+import io.libp2p.core.P2PAbstractChannel
+import io.libp2p.core.PeerId
+import io.libp2p.core.crypto.PrivKey
+import io.libp2p.core.crypto.marshalPublicKey
+import io.libp2p.core.crypto.unmarshalPublicKey
+import io.libp2p.core.multistream.Mode
+import io.libp2p.core.multistream.ProtocolMatcher
+import io.libp2p.core.security.SecureChannel
+import io.libp2p.etc.SECURE_SESSION
+import io.libp2p.etc.events.SecureChannelFailed
+import io.libp2p.etc.events.SecureChannelInitialized
+import io.libp2p.etc.types.toByteArray
+import io.libp2p.etc.types.toByteBuf
+import io.netty.buffer.ByteBuf
+import io.netty.channel.ChannelHandlerContext
+import io.netty.channel.ChannelInboundHandlerAdapter
+import io.netty.channel.SimpleChannelInboundHandler
+import org.apache.logging.log4j.Level
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.apache.logging.log4j.core.config.Configurator
+import spipe.pb.Spipe
+import java.util.Arrays
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.atomic.AtomicInteger
+
+class NoiseXXSecureChannel(private val localKey: PrivKey) :
+        SecureChannel {
+
+    companion object {
+        const val protocolName = "Noise_XX_25519_ChaChaPoly_SHA256"
+        const val announce = "/noise/$protocolName/0.1.0"
+
+        @JvmStatic
+        var localStaticPrivateKey25519: ByteArray = ByteArray(32)
+
+        init {
+            // initialize static Noise key
+            Noise.random(localStaticPrivateKey25519)
+        }
+    }
+
+    private var logger: Logger
+    private var loggerNameParent: String = NoiseXXSecureChannel::class.java.name + this.hashCode()
```

Comment:
I'm not sure if creating a logger per instance is a common practice, but anyways I would at least separate it with `.` (like `NoiseXXSecureChannel::class.java.name + '.' + this.hashCode()`) so you may configure the log level in config. 
(also relates to the comment below)

Reply:
Good points. When I was using `.` as the separator is that only the last segment is displayed in the output, and I'm tracing the instance hierarchy. I also used a logger per instance since log output shows the logger instance name as a separate field in its output and so I figured I would reuse that part, instead of having the instance id appear in the message segment. I'll follow your suggestion and use the `.` separator but keep the instance hierarchy as part of the instance's logger name.

Reply:
Done

Reply:
>  When I was using . as the separator is that only the last segment is displayed in the output
I believe you just need to change the logging configuration accordingly